### ISSUE TITLE
Fix sonarqube issues

### ISSUE THREAD

**Issue by behrisch** [eclipse/sumo#11766]
The security issues are mainly
- C string handling in the fmi coupling (to be solved by refactoring into C++)
- not crypto hardened randomness in the python part (should be ignored)

Bugs
- mainly in sumopy (maybe exclude from analysis?)

**Comment by namdre** [comment]
there are plenty of bugs of the highest category which are not in sumopy but in the C++ core
(https://sonarcloud.io/project/issues?resolved=false&severities=BLOCKER&types=BUG&id=org.eclipse.sumo)

**Comment by behrisch** [comment]
You are right, this ticket just mentions the things which could be solved easily ;-)
